pages/__app/forgotPassword.tsx

```diff
@@ -0,0 +1,144 @@
+  password: z
+    .string()
+    .min(8, { message: 'Password must be at least 8 characters long' }),
+})
+
+const ForgotPasswordPage: NextPage = () => {
+  const [loading, setLoading] = useState(false)
+
+  const router = useRouter()
+
+  const { signInWithEmailAndPassword } = useAuth()
+  const { user, isLoading: currentUserLoading } = useCurrentUser()
+
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      email: '',
+      password: '',
+    },
+  })
+
+  useEffect(() => {
+    if (user && user.id) {
+      router.replace('/dashboard')
+    }
+  }, [router, user])
+
+  const onSubmit = useCallback(
+    async (values: z.infer<typeof formSchema>) => {
+      setLoading(true)
+      try {
+        toast.loading('Please wait...', { id: 'signin-loading' })
+        if (signInWithEmailAndPassword)
+          await signInWithEmailAndPassword({
+            email: values.email,
+            password: values.password,
+          })
+        toast.success('Signed in successfully', { id: 'signin-loading' })
+      } catch (error: any) {
+        onGraphqlErrorToast(error, 'signin-loading')
+      } finally {
+        setLoading(false)
+      }
+    },
+    [signInWithEmailAndPassword]
+  )
+
+  return (
+    <main className="grid min-h-screen md:grid-cols-2">
+      <section className="flex h-full w-full items-center justify-center px-8 pb-20 pt-12 duration-200 md:px-12 lg:px-16">
+        <div className="max-w-md flex-grow">
+          <div className="flex items-end gap-3">
+            <LucideHeartCrack className="h-10 w-10 text-violet-700" />
+          </div>
+          <div className="mt-4 flex w-full flex-col gap-2">
+            <h1 className="text-2xl font-medium">Fogot password</h1>
+            <p className="text-gray-500">
+              Forgot password? Don&apos;t worry! Enter your email, We will send you a reset password link on you email.
+            </p>
+            <Form {...form}>
+              <form
+                onSubmit={form.handleSubmit(onSubmit)}
+                className="mt-4 flex flex-col gap-4"
+              >
+                <FormField
+                  control={form.control}
+                  name="email"
+                  render={({ field }) => (
+                    <FormItem>
+                      <FormLabel>Email</FormLabel>
+                      <FormControl>
+                        <Input
+                          disabled={loading}
+                          placeholder="Enter your email"
+                          {...field}
+                        />
+                      </FormControl>
+                      <FormMessage />
+                    </FormItem>
+                  )}
+                />
+              
+              
+                <Button
+                  disabled={loading}
+                  type="submit"
+                  className="mt-2 w-full"
+                >
+                  Login
```

Comment:
As this form is used for forgot password, may be we can use **Sumbit** or **Send Reset Password Instructions** (you can refer to other websites) in the button label/text instead of using **Login** ?

Reply:
yea but we also require that functionality if we're using something like "Submit" or "Send Reset Password Instructions" but as in this code we're just letting the user to Login without password. 
So ig this works for now, ofc we'll update it to a good way of handling the forgot password feature in the future.